com.unity.sg2/Editor/GraphUI/ShaderGraphEditorWindow.cs

```diff
@@ -268,6 +269,20 @@ public void HandleGraphLoad(ShaderGraphModel shaderGraphModel)
             ShaderGraphCommands.RegisterCommandHandlers(m_GraphTool, m_PreviewManager);
 
             PreviewCommands.RegisterCommandHandlers(GraphTool, m_PreviewManager, shaderGraphModel, GraphView.Dispatcher, GraphView.GraphViewModel);
+
+            // TODO (Joe): With this, we can remove old calls to DefineNode in places the UI expected nodes to reconcretize.
+            shaderGraphModel.GraphHandler.AddBuildCallback(nodeHandler =>
```

Comment:
Would it be a lot of work to go ahead and just get rid of those old DefineNode() calls in this PR? Or are there other things you've got to get to this sprint etc. it's fine to leave the TODO ofc, just figured if we can address it right away, why not